src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -14,8 +13,13 @@ public static class ConfigurationExtensions
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
-        public static IConfigurationBuilder AddUserSecrets([NotNull]this IConfigurationBuilder configuration)
+        public static IConfigurationBuilder AddUserSecrets(this IConfigurationBuilder configuration)
         {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
```

Comment:
@Eilon Yup I know. I wasn't sure if `nameof(configuration)` would be equal to `"configuration"` or something else.  Not sure how it works in extensions method with the syntactic sugar.  Even if it's `"configuration"` I'm not sure if it's very useful in the error message, or if just throwing a `NullReferenceException` would be a better choice for when `this` is null in extension methods.